pkg/kubelet/cm/dra/claiminfo.go

```diff
@@ -179,11 +186,18 @@ func (cache *claimInfoCache) add(res *ClaimInfo) {
 	cache.claimInfo[res.ClaimName+res.Namespace] = res
 }
 
-func (cache *claimInfoCache) get(claimName, namespace string) *ClaimInfo {
+// getClaimInfo returns a deep copy of the cached claimInfo object
+// for the specified claim name and namespace
+func (cache *claimInfoCache) getClaimInfo(claimName, namespace string) (ClaimInfo, error) {
 	cache.RLock()
 	defer cache.RUnlock()
 
-	return cache.claimInfo[claimName+namespace]
+	claimInfo, ok := cache.claimInfo[claimName+namespace]
+	if !ok {
+		return ClaimInfo{}, fmt.Errorf("unable to get resource for namespace: %s, claim: %s", namespace, claimName)
+	}
+
+	return claimInfo.deepCopy(), nil
```

Comment:
This seems to be is the main difference from what we were doing before. Since this always returns a copy we don't have to worry about accesses to it outside of the cache.

Reply:
Yes, this and the correspondent changes in the manager.go caused by this change.